options.go

```diff
@@ -611,6 +620,41 @@ func parseJwtIssuers(issuers []string, msgs []string) ([]jwtIssuer, []string) {
 	return parsedIssuers, msgs
 }
 
+func parseIPWhitelist(ipWhitelistStrs []string, msgs []string) ([]*net.IPNet, []string) {
+	parsedIPNets := make([]*net.IPNet, 0, len(ipWhitelistStrs))
+
+	for ipIndex, ipStr := range ipWhitelistStrs {
+		if !strings.ContainsRune(ipStr, '/') {
+			ip := net.ParseIP(ipStr)
+			if ip == nil {
+				msgs = append(msgs, fmt.Sprintf(
+					"ip_whitelist[%d] (%s) looks like a IP address, but could not be recognized", ipIndex, ipStr))
+				continue
+			}
+
+			// nolint:gocritic
+			if ip.To4() != nil {
+				ipStr += "/32"
+			} else if ip.To16() != nil {
+				ipStr += "/128"
+			} else {
+				msgs = append(msgs, fmt.Sprintf("ip_whitelist[%d] (%s) address neither IPv4 or IPv6", ipIndex, ipStr))
+				continue
+			}
```

Comment:
This logic is a bit terse, could do with a comment to explain what we are doing here and why we are adding subnets to the end

Reply:
Yeah, I can do that.  Dual support of v4 and v6 is always a little terse in my experience.